ibis/expr/tests/test_lineage.py

```diff
@@ -0,0 +1,171 @@
+import pytest
+import ibis
+import ibis.expr.lineage as lin
+from ibis.tests.util import assert_equal
+
+
+@pytest.fixture
+def con():
+    # make sure this is in the directory where you run py.test
+    return ibis.sqlite.connect('crunchbase.db')
+
+
+@pytest.fixture
+def companies(con):
+    return con.table('companies')
+
+
+@pytest.fixture
+def rounds(con):
+    return con.table('rounds')
+
+
+def test_lineage(companies):
+    # single table dependency
+    funding_buckets = [
+        0, 1000000, 10000000, 50000000, 100000000,  500000000, 1000000000
+    ]
+
+    bucket_names = [
+        '0 to 1m',
+        '1m to 10m',
+        '10m to 50m',
+        '50m to 100m',
+        '100m to 500m',
+        '500m to 1b',
+        'Over 1b',
+    ]
+
+    bucket = (
+        companies.funding_total_usd.bucket(funding_buckets, include_over=True)
+    )
+
+    mutated = companies.mutate(
+        bucket=bucket,
+        status=companies.status.fillna('Unknown')
+    )
+
+    filtered = mutated[
+        (companies.founded_at > '2010-01-01') | companies.founded_at.isnull()
+    ]
+
+    grouped = filtered.group_by(['bucket', 'status']).size()
+
+    joined = grouped.mutate(
+        bucket_name=lambda x: x.bucket.label(bucket_names).fillna('Unknown')
+    )
+
+    results = list(lin.lineage(bucket))
+    expected = [
+        bucket,
+        bucket.op().args[0],
+        bucket.op().args[0].op().args[1]
```

Comment:
These tests are as clear as mud. I need to clean them up and make them like the ones below that show the individual pieces in the expected list.